neuralpredictors/layers/encoders/poisson.py

```diff
@@ -0,0 +1,34 @@
+from torch import nn
+
+
+class PoissonEncoder(nn.Module):
+    def __init__(self, core, readout, elu_offset=0, shifter=None):
+        super().__init__()
+        self.core = core
+        self.readout = readout
+        self.shifter = shifter
+        self.offset = elu_offset
+
+    def forward(
+        self,
+        inputs,
+        targets=None,
+        data_key=None,
+        behavior=None,
+        pupil_center=None,
+        trial_idx=None,
```

Comment:
The trial_idx is a special variant, which is not really needed in most cases. I think it's best omit it here, because in the default shifter, that argument isn't needed and will throw an error in the default shifers.

Reply:
Which shifter is the default shifter? We tried to change the shifters such that they all accept it, but don't have to use it.

Reply:
Yes, Mohammad and I thought that it would be better to let the encoder always pass everything (i.e. pupil_center and trial_idx) and then the shifter itself needs to deal with it. That's why we changed all shifters to accept trial_idx. Do you prefer the encoder dealing with it @KonstantinWilleke?

Reply:
The default shifter is the `MLP`.  It takes one input arg. And the trial idx is used in my shifter like this: I stack the pupil center and trial index, and then pass it as one argument. But that's not the "standard" case of dealing with the shifter. The normal case would be just to pass the eye position to the shifter. So I think the trial_idx is not general enough to be part of neuralpredictors.

Reply:
If we remove the `trial_index` from the encoder, then you will not be able to use the encoder here out of the box and would need to inherit and modify forward function, right? So far we only have two shifter classes, so I think it would not hurt to pass the trial_index as an optional argument and potentially use them in the shifter. You are using it because of the problem we had with the camera movement, right? and that might happen again in the future.

Reply:
Yes true, I'd need to modify the forward function then. But the shifter itself can not deal, at least not at the moment, with more than 1 input. So my encoder does the stacking of pupil_center and trial_idx. And adding this stacking as the default for the poisson encoder might be a bit much, but it possibly also wouldn't hurt. 
The only downside I could imagine is this: The trial_idx might be used by some other mechanic as well. Maybe for behavior, or some other hypothetical case where something changes across the experiment. So if a trial_idx is present, stacking it by default with the eye data might lead to problems.

Reply:
Ahhh now I see, sorry, you have modified the shifters!

Reply:
The stacking is a shifter-specific thing, and that is why it should be dealt with in the shifter itself.
Yes exactly, we also changed the shifters. So the encoder and shifter in this PR should work together out of the box, theoretically.

Reply:
thanks, yes I like it!

Reply:
ok, so I resolve this?